packages/dappmanager/src/modules/stakerConfig/set/setConsensusClient.ts

```diff
@@ -0,0 +1,212 @@
+  ConsensusClientMainnet,
+  ConsensusClientPrater,
+  Network
+} from "@dappnode/types";
+
+export async function setConsensusClient<T extends Network>({
+  network,
+  feeRecipient,
+  currentConsensusClient,
+  targetConsensusClient,
+  currentConsClientPkg
+}: {
+  network: Network;
+  feeRecipient: string | null;
+  currentConsensusClient?: ConsensusClient<T> | null;
+  targetConsensusClient?: StakerItemOk<T, "consensus">;
+  currentConsClientPkg?: InstalledPackageDataApiReturn;
+}): Promise<void> {
+  await setConsensusClientConfig({
+    network,
+    feeRecipient,
+    currentConsensusClient,
+    targetConsensusClient,
+    currentConsClientPkg
+  });
+  await setConsensusOnDb(network, targetConsensusClient?.dnpName);
+}
+
+async function setConsensusClientConfig<T extends Network>({
+  network,
+  feeRecipient,
+  currentConsensusClient,
+  targetConsensusClient,
+  currentConsClientPkg
+}: {
+  network: Network;
+  feeRecipient: string | null;
+  currentConsensusClient?: ConsensusClient<T> | null;
+  targetConsensusClient?: StakerItemOk<T, "consensus">;
+  currentConsClientPkg?: InstalledPackageDataApiReturn;
+}): Promise<void> {
+  if (!targetConsensusClient?.dnpName) {
+    if (!currentConsensusClient) {
+      // Stop the current consensus client if no option and not current consensus client
+      logs.info(`Not consensus client selected`);
+      if (currentConsClientPkg)
+        await stopAllPkgContainers(currentConsClientPkg);
+    } else if (!targetConsensusClient?.dnpName && currentConsensusClient) {
+      // Stop the current consensus client if no target provided
+      logs.info(`Not consensus client selected`);
+      if (currentConsClientPkg)
+        await stopAllPkgContainers(currentConsClientPkg);
+    }
+    // Return if no consensus client is selected
+    return;
+  }
+
+  // User settings object: GRAFFITI, FEE_RECIPIENT_ADDRESS, CHECKPOINTSYNC
+  const userSettings: UserSettingsAllDnps = getConsensusUserSettings({
+    dnpName: targetConsensusClient.dnpName,
+    network,
+    feeRecipient: feeRecipient || "",
+    useCheckpointSync: targetConsensusClient.useCheckpointSync
+  });
+
+  if (targetConsensusClient.dnpName && !currentConsensusClient) {
+    const targetConsClientPkg = await listPackageNoThrow({
+      dnpName: targetConsensusClient.dnpName
+    });
+    if (!targetConsClientPkg) {
+      // Install new consensus client if not installed
+      await packageInstall({
+        name: targetConsensusClient.dnpName,
+        userSettings
+      });
+    } else {
+      // Update env if needed
+      await updateConsensusEnv({
+        targetConsensusClient,
+        userSettings
+      });
+      // Start new consensus client if not running
+      await dockerComposeUpPackage(
+        { dnpName: targetConsClientPkg.dnpName },
+        {},
+        {},
+        true
+      );
+    }
+  } else if (targetConsensusClient.dnpName === currentConsensusClient) {
+    if (!currentConsClientPkg) {
```

Comment:
It is already handled, and in that case it will be installed
see 
```typescript
 else if (targetConsensusClient.dnpName !== currentConsensusClient) {
    const targetExecClientPkg = await listPackageNoThrow({
      dnpName: targetConsensusClient.dnpName
    });
    if (!targetExecClientPkg) {
 ...
```